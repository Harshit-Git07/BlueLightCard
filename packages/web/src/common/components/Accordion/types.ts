export interface AccordionProps {
  title: string;
  children: React.ReactNode;
  onClickOpen?: () => void;
}
