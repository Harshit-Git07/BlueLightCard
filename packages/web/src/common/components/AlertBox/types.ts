export interface AlertBoxProps {
  alertType: string;
  title: string;
  description: any;
}

export interface style {
  textColor: string;
  backgroundColor: string;
}

export interface config {
  [key: string]: style;
}
