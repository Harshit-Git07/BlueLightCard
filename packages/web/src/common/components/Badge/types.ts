export type BgColorString = `bg-${string}`;

export type BadgeProps = {
  label: string;
  color: BgColorString;
  size: 'small' | 'large';
};
