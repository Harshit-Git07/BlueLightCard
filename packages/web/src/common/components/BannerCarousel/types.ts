import { CampaignCardProps } from '../CampaignCard/types';

export type BannerCarouselPropers = {
  banners: CampaignCardProps[];
};
