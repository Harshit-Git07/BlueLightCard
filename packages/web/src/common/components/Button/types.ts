import { ThemeVariant } from '@/types/theme';
import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { MouseEventHandler, PropsWithChildren } from 'react';

export type ButtonProps = PropsWithChildren & {
  type?: 'button' | 'submit' | 'link';
  disabled?: boolean;
  slim?: boolean;
  className?: string;
  variant?: ThemeVariant;
  invertColor?: boolean;
  iconLeft?: IconDefinition;
  iconRight?: IconDefinition;
  href?: string;
  onClick?: MouseEventHandler;
};
