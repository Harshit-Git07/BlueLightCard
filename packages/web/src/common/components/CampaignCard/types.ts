export type CampaignCardProps = {
  name: string;
  image: string;
  linkUrl: string;
};
