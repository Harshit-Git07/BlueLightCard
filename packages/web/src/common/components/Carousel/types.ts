export type CarouselProps = {
  children: React.ReactNode[] | React.ReactNode;

  // Display
  hidePillButtons?: boolean;
  loop?: boolean;
  focusCenter?: boolean;

  // Breakpoints
  elementsPerPageDesktop?: number;
  elementsPerPageLaptop?: number;
  elementsPerPageTablet?: number;
  elementsPerPageMobile?: number;

  // Control
  clickToScroll?: boolean;
  showControls?: boolean;
  swipeToScroll?: boolean;

  // Auto play
  autoPlay?: boolean;
  autoPlayIntervalMs?: number;

  // Custom styles
  className?: string;
  focusedClassName?: string;
};
