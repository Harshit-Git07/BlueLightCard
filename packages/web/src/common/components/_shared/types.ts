import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { ChangeEventHandler, ForwardedRef, MouseEventHandler, PropsWithChildren } from 'react';

export type InputFieldWrapperProps = PropsWithChildren & {
  icon?: IconDefinition;
  iconRight?: IconDefinition;
  showRightIcon?: boolean;
  showErrorState?: boolean;
  select?: boolean;
  showSuccessState?: boolean;
  onRightIconClick?: MouseEventHandler;
};

export interface InputFieldSharedProps<E, V = string> {
  name?: string;
  error?: boolean;
  success?: boolean;
  value?: V;
  required?: boolean;
  _ref?: ForwardedRef<unknown>;
  onChange?: ChangeEventHandler<E>;
}
