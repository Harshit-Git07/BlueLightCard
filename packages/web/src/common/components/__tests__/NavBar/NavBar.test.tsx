import { render, screen } from '@testing-library/react';
import NavBar from '../../NavBar/NavBar';
import { NavigationItem } from '../../NavBar/types';
import { BRANDS } from '../../../types/brands.enum';

jest.mock('../../NavBar/helpers/getNavigationItems', () => {
  return {
    getNavigationItems: (brand: BRANDS, isAuthenticated: boolean) => [
      { id: 'mockNavItem', label: 'Mock Label' },
    ],
  };
});

describe('Navbar', () => {
  it('renders the authenticatedNavBar correctly when isAuthenticated is true', () => {
    const { container } = render(
      <NavBar
        isAuthenticated
        onSearchCategoryChange={jest.fn()}
        onSearchCompanyChange={jest.fn()}
        onSearchTerm={jest.fn()}
      />
    );
    expect(container).toMatchSnapshot();
    expect(screen.getByTestId('authenticated-navbar')).toBeDefined();
  });
  it('renders the unauthenticated header when isAuthenticated if false', () => {
    const { container } = render(
      <NavBar
        isAuthenticated={false}
        onSearchCategoryChange={jest.fn()}
        onSearchCompanyChange={jest.fn()}
        onSearchTerm={jest.fn()}
      />
    );
    expect(container).toMatchSnapshot();
    expect(screen.getByTestId('unauthenticated-navbar')).toBeDefined();
  });
});
