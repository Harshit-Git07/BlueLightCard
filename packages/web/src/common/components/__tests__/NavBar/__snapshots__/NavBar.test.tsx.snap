// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders the authenticatedNavBar correctly when isAuthenticated is true 1`] = `
<div>
  <header
    class="sticky top-0 z-10"
  >
    <div
      data-testid="authenticated-navbar"
    >
      <div
        class="dark:bg-NavBar-bg-colour-dark bg-NavBar-bg-colour text-NavBar-item-text-colour dark:text-NavBar-item-text-colour-dark "
      >
        <div
          class="h-[72px] px-4 tablet:px-14 laptop:container laptop:mx-auto flex justify-between items-center"
        >
          <a
            aria-label="Navigate Home"
            class="h-[35px] tablet:h-[42px] laptop:h-[55px] hover:opacity-100"
            data-testid="brandLogo"
            href="/members-home"
          >
            <img
              alt="Navigate Home"
              class="fill-logo-colour dark:fill-logo-colour-dark h-full"
            />
          </a>
          <div
            class="flex gap-7 h-full items-center"
          >
            <div
              class="hidden desktop:flex desktop:h-full"
            >
              <div
                class="flex flex-1 gap-5 h-full justify-between"
              >
                <div
                  class="font-['MuseoSans'] text-xs font-semibold cursor-pointer text-colour-primary dark:text-colour-primary-dark my-1"
                  data-testid="on-click-link"
                >
                  Mock Label
                </div>
              </div>
            </div>
            <div
              class="flex gap-5 items-center"
            >
              <div
                class="hidden tablet:block tablet:w-[343px] w-full"
              >
                <form
                  class="px-2 z-10 w-full py-4 "
                >
                  <div
                    class="relative"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-magnifying-glass fa-xl absolute left-2 px-2 top-1/2 transform -translate-y-1/2 text-searchBar-icon-colour-light dark:text-searchBar-icon-colour-dark"
                      data-icon="magnifying-glass"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
                        fill="currentColor"
                      />
                    </svg>
                    <input
                      autocomplete="off"
                      class="search-bar text-ellipsis pl-14 pr-12 py-3 rounded-full w-full overflow-x-hidden bg-searchBar-bg-colour-light dark:bg-searchBar-bg-colour-dark focus:outline-none text-searchBar-label-colour-light dark:text-searchBar-label-colour-dark dark:border-searchBar-outline-colour-dark font-searchBar-label-font text-searchBar-label-font font-searchBar-label-font-weight tracking-searchBar-label-font leading-searchBar-label-font
          border border-searchBar-outline-colour-light"
                      enterkeyhint="search"
                      id="searchInput"
                      placeholder="Search for offers or companies"
                      type="search"
                      value=""
                    />
                  </div>
                </form>
              </div>
              <a
                aria-label="Notifications"
                class="flex h-full justify-center items-center gap-1 cursor-pointer text-NavBar-icon-colour hover:text-NavBar-link-hover-colour dark:text-NavBar-icon-colour-dark dark:hover:text-NavBar-link-hover-colour-dark"
                data-testid="anchor-link"
                href="/notifications.php"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-bell "
                  data-icon="bell"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224 0c-17.7 0-32 14.3-32 32V51.2C119 66 64 130.6 64 208v18.8c0 47-17.3 92.4-48.5 127.6l-7.4 8.3c-8.4 9.4-10.4 22.9-5.3 34.4S19.4 416 32 416H416c12.6 0 24-7.4 29.2-18.9s3.1-25-5.3-34.4l-7.4-8.3C401.3 319.2 384 273.9 384 226.8V208c0-77.4-55-142-128-156.8V32c0-17.7-14.3-32-32-32zm45.3 493.3c12-12 18.7-28.3 18.7-45.3H224 160c0 17 6.7 33.3 18.7 45.3s28.3 18.7 45.3 18.7s33.3-6.7 45.3-18.7z"
                    fill="currentColor"
                  />
                </svg>
              </a>
              <button
                class="flex h-full justify-center items-center gap-1 cursor-pointer tablet:hidden text-NavBar-icon-colour dark:text-NavBar-icon-colour-dark"
                data-testid="mobile-nav-toggle-button"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-bars "
                  data-icon="bars"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
                    fill="currentColor"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          class="h-[72px] hidden tablet:px-14 tablet:flex desktop:hidden desktop:mx-auto"
        >
          <div
            class="flex flex-1 gap-5 h-full justify-between"
          >
            <div
              class="font-['MuseoSans'] text-xs font-semibold cursor-pointer text-colour-primary dark:text-colour-primary-dark my-1"
              data-testid="on-click-link"
            >
              Mock Label
            </div>
          </div>
        </div>
      </div>
      <div
        class="tablet:hidden bg-colour-surface dark:bg-colour-surface-dark"
      >
        <form
          class="px-2 z-10 w-full py-4 "
        >
          <div
            class="relative"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-magnifying-glass fa-xl absolute left-2 px-2 top-1/2 transform -translate-y-1/2 text-searchBar-icon-colour-light dark:text-searchBar-icon-colour-dark"
              data-icon="magnifying-glass"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
                fill="currentColor"
              />
            </svg>
            <input
              autocomplete="off"
              class="search-bar text-ellipsis pl-14 pr-12 py-3 rounded-full w-full overflow-x-hidden bg-searchBar-bg-colour-light dark:bg-searchBar-bg-colour-dark focus:outline-none text-searchBar-label-colour-light dark:text-searchBar-label-colour-dark dark:border-searchBar-outline-colour-dark font-searchBar-label-font text-searchBar-label-font font-searchBar-label-font-weight tracking-searchBar-label-font leading-searchBar-label-font
          border border-searchBar-outline-colour-light"
              enterkeyhint="search"
              id="searchInput"
              placeholder="Search for offers or companies"
              type="search"
              value=""
            />
          </div>
        </form>
      </div>
    </div>
  </header>
</div>
`;

exports[`Navbar renders the unauthenticated header when isAuthenticated if false 1`] = `
<div>
  <header
    class="sticky top-0 z-10"
  >
    <div
      data-testid="unauthenticated-navbar"
    >
      UnauthenticatedNavBar
    </div>
  </header>
</div>
`;
